lib/Data/Validate/Sanctions.pm

```diff
@@ -316,16 +318,20 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
 sub _load_data {
     my $self          = shift;
     my $sanction_file = $self->{sanction_file};
-    $self->{last_time}               //= 0;
+    $self->{last_modification}       //= 0;
+    $self->{last_index}              //= 0;
     $self->{_data}                   //= {};
     $self->{_sanctioned_name_tokens} //= {};
     $self->{_token_sanctioned_names} //= {};
 
     if (-e $sanction_file) {
-        return $self->{_data} if stat($sanction_file)->mtime <= $self->{last_time} && $self->{_data};
-        $self->{last_time} = stat($sanction_file)->mtime;
-        $self->{_data}     = LoadFile($sanction_file);
+        return $self->{_data} if stat($sanction_file)->mtime <= $self->{last_modification} && $self->{_data};
+        $self->{last_modification} = stat($sanction_file)->mtime;
+        $self->{_data}             = LoadFile($sanction_file);
     }
+
+    return $self->{_data} if $self->{last_modification} <= $self->{last_index};
```

Comment:
Do we need this line? if data wasn't modified we return at #328, if we load new file then we always need to update the index... What case are we handling here?